digital_land/update.py

```diff
@@ -1,44 +1,42 @@
 from datetime import datetime, date
```

Comment:
I wonder if some of the functions here might make sense as methods on the Collection class. Or maybe we should encapsulate the management of source.csv and endpoint.csv into CollectionConfig or CollectionCatalog or something. e.g. `CollectionConfig.add_new_source_endpoint` and `Collection.has_collected_resource`, `Collection.get_failing_endpoints` etc.

Reply:
Certainly not a blocker for merging, but if we decide we want to refactor this we could raise a ticket to do the work in future.